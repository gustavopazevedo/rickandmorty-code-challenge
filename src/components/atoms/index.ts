export { default as IconButton } from './IconButton';
export { default as Input } from './Input';
export { default as StatusBadge } from './StatusBadge';
export { default as TableCell } from './TableCell';
