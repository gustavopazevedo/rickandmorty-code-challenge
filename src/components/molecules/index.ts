export { default as CustomTooltipChart } from './CustomTooltipChart';
export { default as InfoCard } from './InfoCard';
export { default as Loader } from './Loader';
export { default as SearchInput } from './SearchInput';
export { default as TableRow } from './TableRow';
