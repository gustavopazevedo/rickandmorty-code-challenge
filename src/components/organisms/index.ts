export { default as AliveCharacterCounter } from './AliveCharacterCounter';
export { default as CharacterTable } from './CharacterTable';
export { default as HumanCharacterCounter } from './HumanCharacterCounter';
export { default as LocationChart } from './LocationChart';
export { default as LocationCounter } from './LocationCounter';
export { default as Section } from './Section';
