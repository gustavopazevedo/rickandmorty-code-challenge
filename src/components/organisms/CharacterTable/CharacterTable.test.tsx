import { render } from '@testing-library/react';

import { CharacterTable } from '@/components/organisms';

jest.mock('next/navigation', () => ({
  useRouter: jest.fn(),
  useSearchParams: () => ({
    get: () => '',
  }),
  usePathname: jest.fn(),
}));

// Mock your custom useCharacters hook
jest.mock('@/hooks', () => ({
  useCharacters: () => ({
    characters: [
      {
        id: '1',
        name: 'Rick Sanchez',
        status: 'Alive',
        species: 'Human',
        gender: 'Male',
        image: 'https://rickandmortyapi.com/api/character/avatar/1.jpeg',
        location: {
          name: 'Citadel of Ricks',
          __typename: 'Location',
        },
        __typename: 'Character',
      },
      {
        id: '2',
        name: 'Morty Smith',
        status: 'Alive',
        species: 'Human',
        gender: 'Male',
        image: 'https://rickandmortyapi.com/api/character/avatar/2.jpeg',
        location: {
          name: 'Citadel of Ricks',
          __typename: 'Location',
        },
        __typename: 'Character',
      },
    ],
    loading: false,
    error: null,
    nextPage: null,
    fetchMore: jest.fn(),
  }),
}));

describe('CharacterTable Snapshot', () => {
  beforeEach(() => {
    const mockIntersectionObserver = jest.fn();

    mockIntersectionObserver.mockReturnValue({
      observe: () => null,
      unobserve: () => null,
      disconnect: () => null,
    });
    window.IntersectionObserver = mockIntersectionObserver;
  });

  it('should match snapshot', () => {
    const { container } = render(<CharacterTable />);
    expect(container).toMatchSnapshot();
  });
});
