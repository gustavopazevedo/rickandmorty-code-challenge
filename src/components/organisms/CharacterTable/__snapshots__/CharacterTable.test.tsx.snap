// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`CharacterTable Snapshot should match snapshot 1`] = `
<div>
  <div
    class="relative"
  >
    <div
      class="w-screen-minus-16 relative h-[400px] overflow-auto rounded-lg border border-gray-300 lg:h-[600px] lg:w-full"
    >
      <table
        class="relative w-[600px] table-auto text-left text-sm sm:w-full"
      >
        <thead
          class="sticky top-0 z-10 bg-gray-100"
        >
          <tr
            class=""
          >
            <th
              class="p-2 text-left font-semibold text-gray-700 lg:p-4"
            >
              Character
            </th>
            <th
              class="p-2 text-left font-semibold text-gray-700 lg:p-4"
            >
              Status
            </th>
            <th
              class="p-2 text-left font-semibold text-gray-700 lg:p-4"
            >
              Species
            </th>
            <th
              class="p-2 text-left font-semibold text-gray-700 lg:p-4"
            >
              Gender
            </th>
            <th
              class="p-2 text-left font-semibold text-gray-700 lg:p-4"
            >
              Location
            </th>
          </tr>
        </thead>
        <tbody
          class="divide-y divide-gray-200"
        >
          <tr
            class="odd:bg-white even:bg-gray-50 hover:bg-gray-100"
          >
            <td
              class="p-2 lg:p-4 min-w-[200px]"
            >
              <div
                class="flex items-center gap-4"
              >
                <img
                  alt="Rick Sanchez"
                  class="rounded-full"
                  data-nimg="1"
                  decoding="async"
                  height="40"
                  loading="lazy"
                  src="/_next/image?url=https%3A%2F%2Frickandmortyapi.com%2Fapi%2Fcharacter%2Favatar%2F1.jpeg&w=96&q=75"
                  srcset="/_next/image?url=https%3A%2F%2Frickandmortyapi.com%2Fapi%2Fcharacter%2Favatar%2F1.jpeg&w=48&q=75 1x, /_next/image?url=https%3A%2F%2Frickandmortyapi.com%2Fapi%2Fcharacter%2Favatar%2F1.jpeg&w=96&q=75 2x"
                  style="color: transparent;"
                  width="40"
                />
                <span
                  class="font-medium text-gray-900"
                >
                  Rick Sanchez
                </span>
              </div>
            </td>
            <td
              class="p-2 lg:p-4"
            >
              <span
                class="rounded-full px-3 py-1 text-xs font-semibold bg-green-100 text-green-800"
              >
                ALIVE
              </span>
            </td>
            <td
              class="p-2 lg:p-4"
            >
              Human
            </td>
            <td
              class="p-2 lg:p-4"
            >
              Male
            </td>
            <td
              class="p-2 lg:p-4"
            >
              Citadel of Ricks
            </td>
          </tr>
          <tr
            class="odd:bg-white even:bg-gray-50 hover:bg-gray-100"
          >
            <td
              class="p-2 lg:p-4 min-w-[200px]"
            >
              <div
                class="flex items-center gap-4"
              >
                <img
                  alt="Morty Smith"
                  class="rounded-full"
                  data-nimg="1"
                  decoding="async"
                  height="40"
                  loading="lazy"
                  src="/_next/image?url=https%3A%2F%2Frickandmortyapi.com%2Fapi%2Fcharacter%2Favatar%2F2.jpeg&w=96&q=75"
                  srcset="/_next/image?url=https%3A%2F%2Frickandmortyapi.com%2Fapi%2Fcharacter%2Favatar%2F2.jpeg&w=48&q=75 1x, /_next/image?url=https%3A%2F%2Frickandmortyapi.com%2Fapi%2Fcharacter%2Favatar%2F2.jpeg&w=96&q=75 2x"
                  style="color: transparent;"
                  width="40"
                />
                <span
                  class="font-medium text-gray-900"
                >
                  Morty Smith
                </span>
              </div>
            </td>
            <td
              class="p-2 lg:p-4"
            >
              <span
                class="rounded-full px-3 py-1 text-xs font-semibold bg-green-100 text-green-800"
              >
                ALIVE
              </span>
            </td>
            <td
              class="p-2 lg:p-4"
            >
              Human
            </td>
            <td
              class="p-2 lg:p-4"
            >
              Male
            </td>
            <td
              class="p-2 lg:p-4"
            >
              Citadel of Ricks
            </td>
          </tr>
          <tr
            class=""
          >
            <td
              class="p-2 lg:p-4 col-span-4 w-full text-center"
              colspan="4"
            >
              No more characters to load
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
